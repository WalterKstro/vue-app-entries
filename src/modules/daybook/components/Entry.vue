<template>
  <div class="card mb-2" @click="gotoEntry">
    <div class="card-body">
      <h5 class="card-title">{{entry.title}}</h5>
      <p class="card-text m-0">{{oneStringExtractOfDescription}}</p>
      <p class="card-text text-end"><small class="text-muted">{{entry.date}}</small></p>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    entry: {
      type: Object,
      required: true,
    },
  },
  methods: {
    gotoEntry(){
      this.$router.push({ name: 'entry', params: { id: this.entry.id } })
    }
  },
  computed: {
    oneStringExtractOfDescription(){
      const arrayOfWordsOfExtract = this.entry.description.split(' ')

      if(arrayOfWordsOfExtract.length > 15) arrayOfWordsOfExtract.length = 15
      
      return arrayOfWordsOfExtract.join(' ').concat('...')
    }
  },
}
</script>
<style lang="scss" scoped>
.card{
  &:hover{
    background-color: #f5f5f5;
    cursor: pointer;
  }
}
</style>
