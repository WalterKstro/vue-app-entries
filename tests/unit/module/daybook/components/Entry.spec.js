import { shallowMount } from '@vue/test-utils';
import Entry from '../../../../../src/modules/daybook/components/Entry.vue';

describe('Pruebas en el componente <Entry/>', () => {

    let wrapper;
    let props;
    let mockRouter;
    beforeEach(() => {
        
        props = {
            entry : {
                id: '-MyIm6N1MD0hlNXSBT9O',
                date:"miÃ©rcoles, 16 de marzo de 2022",
                description :"The product itself is focused on data collection; from the first 50 companies ESGgo spoke to, they learned that there were no tools available.",
                title:"The Intuitive Vue Framework"
            }
        }

        mockRouter = {
            push: jest.fn()
        }

        wrapper = shallowMount(Entry, {
            props,
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        });
    })

    test('Debe mostrar el componente, correctamente', () => {
        expect( wrapper.html() ).toMatchSnapshot();
    })
    test('Debe retornar un extracto, la propiedad computada oneStringExtractOfDescription de 15 palabras', () => {
        const extract = wrapper.vm.oneStringExtractOfDescription.split(' ');
        expect( extract.length ).toBe(15);
    })
    test('Debe redireccionar a la entrada, si se hace clic', () => {
        wrapper.find('.card').trigger('click');

        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'entry', params: { id: props.entry.id } });
    })
})