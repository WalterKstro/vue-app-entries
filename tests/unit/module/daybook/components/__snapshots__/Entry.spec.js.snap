// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Entry/> Debe mostrar el componente, correctamente 1`] = `
"<div class=\\"card mb-2\\">
  <div class=\\"card-body\\">
    <h5 class=\\"card-title\\">The Intuitive Vue Framework</h5>
    <p class=\\"card-text m-0\\">The product itself is focused on data collection; from the first 50 companies ESGgo spoke...</p>
    <p class=\\"card-text text-end\\"><small class=\\"text-muted\\">mi√©rcoles, 16 de marzo de 2022</small></p>
  </div>
</div>"
`;
